Tests/Fuzzers/FuzzTarget.cpp

```diff
@@ -1,87 +1,76 @@
-#include <iostream>
-
-#include <IPv4Layer.h>
-#include <Packet.h>
 #include <PcapFileDevice.h>
+#include <Packet.h>
+#include <Logger.h>
+#include "DumpToFile.h"
+#include "ReadParsedPacket.h"
 
-#include "Logger.h"
-
-#define TMP_FILEPATH "/tmp/fuzz_sample.pcap"
+static std::string tmpName;
+static std::string tmpFile;
 
-// This function is created as PcapPlusPlus doesn't seem to offer a way of
-// parsing Pcap files directly from memory
-int dumpDataToPcapFile(const uint8_t *data, size_t size)
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 {
-	FILE *fd;
-	int written = 0;
+	if (tmpName.empty())
+		tmpName = tmpnam (NULL);
 
-	fd = fopen(TMP_FILEPATH, "wb");
-	if (fd == NULL)
-	{
-		std::cerr << "Error opening pcap file for writing\n";
-		return -1;
-	}
+	if (tmpFile.empty())
+		tmpFile = tmpName + FILE_EXT;
 
-	written = fwrite(data, 1, size, fd);
-	if (static_cast<size_t>(written) != size)
+	if (dumpDataToPcapFile(data, size, tmpFile.c_str()) != 0)
 	{
-		std::cerr << "Error writing pcap file\n";
-		fclose(fd);
+		std::cerr << "Can't Dump buffer to the '" << tmpFile << "' file!!!!\n";
 		return -1;
 	}
 
-	fclose(fd);
-
-	return 0;
-}
+	pcpp::Logger::getInstance().suppressLogs();
 
-extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size)
-{
-	if (dumpDataToPcapFile(Data, Size) < 0)
+	std::unique_ptr<pcpp::IFileReaderDevice> reader(pcpp::IFileReaderDevice::getReader(tmpFile));
+	if (!reader->open())
 	{
-		std::cerr << "Can't Dump buffer to a PCAP file!!!!\n";
-		return 0;
+		std::cerr << "Error opening the '" << tmpFile << "' file\n";
+		return -1;
 	}
 
-	// Disable logs
-	pcpp::Logger::getInstance().suppressLogs();
+	pcpp::IPcapDevice::PcapStats stats;
+	reader->getStatistics(stats);
+	std::cout << "Read " << stats.packetsRecv << " packets successfully and "
+	                     << stats.packetsDrop << " packets could not be read" << std::endl;
 
-	// open a pcap file for reading
-	pcpp::PcapFileReaderDevice reader(TMP_FILEPATH);
-	if (!reader.open())
+	if (auto ngReader = dynamic_cast<pcpp::PcapNgFileReaderDevice*>(reader.get()))
 	{
-		std::cerr << "Error opening the pcap file\n";
-		return 0;
+		std::cout << "OS is '" << ngReader->getOS() << "'; Hardware is '" << ngReader->getHardware() << "'"
+				  << "'; CaptureApplication is '" << ngReader->getCaptureApplication()
+				  << "'; CaptureFileComment is '" << ngReader->getCaptureFileComment()
+				  << "'" << std::endl;
 	}
 
-	// read the first (and only) packet from the file
-	pcpp::RawPacket rawPacket;
-	if (!reader.getNextPacket(rawPacket))
+	pcpp::RawPacketVector packets;
+	if (reader->getNextPackets(packets, 1) != 1)
 	{
 		std::cerr << "Couldn't read the first packet in the file\n";
 		return 0;
 	}
 
+	pcpp::RawPacket& rawPacket = *packets.front();
```

Comment:
Why do we need to call `reader->getNextPackets()` before calling `reader.getNextPacket()` in a loop?

Reply:
It is just a little trick to increase variety of functions that the fuzzer calls to increase the coverage.